### PR TITLE
[dca] filter pod annotation during workloadmeta collection

### PR DESCRIPTION
**Pull request by clamoriniere**
### What does this PR do?

Add an option in cluster-agent to filter Pods annotations during workloadmeta-store
collection.

### Motivation

It allows to reduce the cluster-agent memory usage when `cluster_agent.collect_kubernetes_tags`
is enabled.

### Describe how to test/QA your changes

To ease testing it could be good to wait for `cluster-agent workload-list` command it
available in the cluster-agent (it should also be part of 7.43.0 version too).

* check the output of `cluster-agent workload-list` and if the annotations are properly filtered
* set the option `cluster_agent.workloadmeta.pod_annotations_exclude` with other annotation keys.
  and verify that the new values are used. 

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/cluster-agent-workload-pod-annotations-exclude-30b151c00df6b5fd.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+enhancements:
+  - |
+    To reduce "cluster-agent" memory consomption when `cluster_agent.collect_kubernetes_tags`
```
Comment (maycmlee) [inline releasenotes/notes/cluster-agent-workload-pod-annotations-exclude-30b151c00df6b5fd.yaml:4]:
```suggestion
    To reduce "cluster-agent" memory consumption when the `cluster_agent.collect_kubernetes_tags`
```

File: releasenotes/notes/cluster-agent-workload-pod-annotations-exclude-30b151c00df6b5fd.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+    option is enabled, we introduce `cluster_agent.workloadmeta.pod_annotations_exclude` option
```
Comment (maycmlee) [inline releasenotes/notes/cluster-agent-workload-pod-annotations-exclude-30b151c00df6b5fd.yaml:5]:
```suggestion
    option is enabled, the `cluster_agent.workloadmeta.pod_annotations_exclude` option has been added
```

File: releasenotes/notes/cluster-agent-workload-pod-annotations-exclude-30b151c00df6b5fd.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+    to exclude Pod annotation from the extracted Pod metadata.
```
Comment (maycmlee) [inline releasenotes/notes/cluster-agent-workload-pod-annotations-exclude-30b151c00df6b5fd.yaml:6]:
```suggestion
    for excluding pod annotation from the extracted pod metadata.
```

**Review by maycmlee — COMMENTED** [review]
Left some feedback

**Review by maycmlee — APPROVED** [review]

**Review by Kaderinho — APPROVED** [review]

File: pkg/config/config.go [hunk 4]
```diff
@@ -716,6 +716,10 @@ func InitConfig(config Config) {
 	config.BindEnvAndSetDefault("cluster_agent.max_leader_connections", 100)
 	config.BindEnvAndSetDefault("cluster_agent.client_reconnect_period_seconds", 1200)
 	config.BindEnvAndSetDefault("cluster_agent.collect_kubernetes_tags", false)
+	config.BindEnvAndSetDefault("cluster_agent.kubernetes_resources_collection.pod_annotations_exclude", []string{
+		"kubectl.kubernetes.io/last-applied-configuration",
+		"^ad.datadoghq.com\\/([[:alnum:]].*.)?(checks|check_names|init_configs|instances)$",
```
Comment (L3n41c) [inline pkg/config/config.go:721]: Dots are matching any character. In order to match a dot, it needs to be escaped.

```suggestion
		`^kubectl\.kubernetes\.io/last-applied-configuration$`,
		`^ad\.datadoghq\.com/([[:alnum:]]+\.)?(checks|check_names|init_configs|instances)$`,
```

**Review by L3n41c — APPROVED** [review]

File: pkg/config/config.go [hunk 5]
```diff
@@ -716,6 +716,10 @@ func InitConfig(config Config) {
 	config.BindEnvAndSetDefault("cluster_agent.max_leader_connections", 100)
 	config.BindEnvAndSetDefault("cluster_agent.client_reconnect_period_seconds", 1200)
 	config.BindEnvAndSetDefault("cluster_agent.collect_kubernetes_tags", false)
+	config.BindEnvAndSetDefault("cluster_agent.kubernetes_resources_collection.pod_annotations_exclude", []string{
+		`^kubectl\.kubernetes\.io/last-applied-configuration$`,
+		`^ad\.datadoghq\.com/([[:alnum:]]+\.)?(checks|check_names|init_configs|instances)$`,
```
Comment (clamoriniere) [inline pkg/config/config.go:721]:
```suggestion
		`^kubectl\.kubernetes\.io\/last-applied-configuration$`,
		`^ad\.datadoghq\.com\/([[:alnum:]]+\.)?(checks|check_names|init_configs|instances)$`,
```

**Pull request merged**